app/Providers/AppServiceProvider.php

```diff
@@ -22,30 +22,6 @@ public function register(): void {}
      */
     public function boot(): void
     {
-        // @codeCoverageIgnoreStart
-        UploadedFile::macro('makeFromUrl', function (string $url): ?UploadedFile {
-            $tempFile = tempnam(sys_get_temp_dir(), Str::random(32));
-
-            if ($tempFile === false) {
-                return null;
-            }
-
-            $file = file_get_contents($url);
-
-            if ($file === false) {
-                return null;
-            }
-
-            file_put_contents($tempFile, $file);
-
-            return new UploadedFile(
-                $tempFile,
-                basename($url),
-                mime_content_type($tempFile) ?: null,
-                null,
-                true
-            );
-        });
-        // @codeCoverageIgnoreEnd
+        UploadedFile::macro('makeFromUrl', fn (string $url): ?UploadedFile => (new FileUploaderFromUrl)($url));
```

Comment:
Personally I'd use the container here. Makes it easier to mock in other tests should you need to.

```suggestion
        UploadedFile::macro('makeFromUrl', function (string $url): ?UploadedFile { 
            $fileUploaderFromUrl = App::make(FileUploaderFromUrl::class);
            return $fileUploaderFromUrl($url)
        });
```

Reply:
I'll think on it later, merging it for now!